PowerShell/JumpCloud.Office365.SSO/Public/Enable-JumpCloud.Office365.SSO.ps1

```diff
@@ -21,53 +19,51 @@ function Enable-JumpCloud.Office365.SSO {
+            $DirectorySynchronizationEnabled = Get-MgOrganization -ErrorAction SilentlyContinue -ErrorVariable ProcessError | Select-Object OnPremisesSyncEnabled
 
             if ($ProcessError) {
-                Connect-MsolService
-                $DirectorySynchronizationEnabled = Get-MsolCompanyInformation | Select-Object DirectorySynchronizationEnabled
+                Connect-MgGraph -Scopes "Domain.ReadWrite.All"
+                $DirectorySynchronizationEnabled = Get-MgOrganization | Select-Object OnPremisesSyncEnabled
             }
 
-            $MSDomains = Get-MSOnlineDomains
+            $MSDomains = Get-MgGraphDomains
 
             if ($MSDomains.($Domain) -eq $null) {
                 Write-Warning  "Typo? $Domain is not a valid domain within your Office365 tenant"
-                Write-Host "To see a list of valid domains in your Office 365 run the command 'Get-MsolDomain'" -ForegroundColor Green
+                Write-Host "To see a list of valid domains in your Office 365 run the command 'Get-MgDomain'" -ForegroundColor Green
                 Write-Host "Update your JumpCloud Office 365 SSO connector with the valid domain, download the XML metadata and try again!" -ForegroundColor Green
                 Return
             }
 
             if ( $DirectorySynchronizationEnabled -eq $true) {
-                Write-Warning  "Directory Synchronization is enabled run the command:'Set-MsolDirSyncEnabled -EnableDirSync $false' to disable and try again"
+                Write-Warning  "Directory Synchronization is enabled run the command:'Update-MgOrganization -OrganizationId (Get-MgOrganization).Id -BodyParameter @{onPremisesSyncEnabled = $false}' to disable and try again"
             }
 
             else {
-
                 $SetDomainParams = @{
                     DomainName                      = $Domain
-                    FederationBrandName             = $Domain
-                    Authentication                  = "Federated"
-                    IssuerUri                       = $Domain
-                    LogOffUri                       = $logoutUrl
-                    PassiveLogOnUri                 = $IDPUrl
-                    ActiveLogOnUri                  = $idpUrl
+                    DisplayName                     = $Domain
+                    IssuerUri                       = "https://$Domain"
+                    SignOutUri                      = $logoutUrl
+                    PassiveSignInUri                = $IDPUrl
+                    ActiveSignInUri                 = $idpUrl
                     SigningCertificate              = $certificate
-                    PreferredAuthenticationProtocol = "SAMLP"
-
+                    PreferredAuthenticationProtocol = "saml"
+                    federatedIdpMfaBehavior         = "acceptIfMfaDoneByFederatedIdp"
```

Comment:
Without changing this to "SAML" I never was able to use the module to enable authentication. The `federatedIdpMfaBehavior` property also seems to be required